### PR TITLE
Acceptance testing with Ape; Relocate contracts to `nucypher-contracts`

### PR DESCRIPTION
**Pull request by KPrasch**
##### what this does

- mostly closes https://github.com/nucypher/nucypher/issues/3055
- follow-up https://github.com/nucypher/nucypher-contracts/pull/66
- ports blockchain tests to Ape (`ape test`)
- removes all solidity source code and contract tests (relocated to `nucypher/nucypher-contracts`)
- adapts Testerchain fixture to handle ape-managed `EthereumTester` as a provider
- introduces `InMemoryContractRegistry.from_ape_deployments`

#####  Removals

- `tests.acceptance.blockchain.eth.sol` package
    - `source/` (solidity source code)
    -  `compile/` (solidity compilation tools)
- `blockchain.eth.deployers` module and `Deployers`
- `blockchain.eth.actors.ContractAdministrator`
- `blockchain.eth.interfaces.BlockchainDeployerInterface`
- `cli.commands.deploy` and supporting `nucypher-deploy` CLI functionality

#### todo

- [x] configure ape for local & remote source compilation
- [x] integrate ape compilation & deployments into nucypher pytest fixtures
- [x] importing ethpm sandard package JSON to contract registries
- [x] make remaining tests pass
    - [x] unit
    - [x] integration
    - [x] acceptance
        - [x] blockchain (ape project)
        - [x] everything else
    
##### follow-ups needed

- [ ] add `eth-ape` to dependencies and relock
    - [ ] remove direct project dependency on `web3` or other dependencies of `eth-ape` in `Pipfile`
- [ ] integrate into CI
    - [x] set `ape-config.yaml` to use `github: nucypher/nucypher-contracts` OR
    - ~~[ ] checkout a local copy of nucypher-contracts on CI and test against it (`local: /path/to/nucypher-contracts`)~~
    - [ ] determine a good default nucypher-contracts branch, ref, or tag to test against `ref: ...`
- [x] ~~install ape as a pytest plugin for debugging support~~ (launch via `pytest -s -m ape_test`)
- [ ] update contributing documentation

##### Co-Authors

@derekpierre 
@KPrasch

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -31,6 +31,10 @@
 from tests.mock.io import MockStdinWrapper
 
 
+def pytest_addhooks(pluginmanager):
+    pluginmanager.set_blocked('ape_test')
```
Comment (derekpierre) [inline tests/integration/conftest.py:35]: 🔥

**Review by KPrasch — COMMENTED** [review]
approved, though I am only a co-author 🤠

File: tests/utils/blockchain.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import os
+from ape_test import LocalProvider
```
Comment (jMyles) [inline tests/utils/blockchain.py:2]: This doesn't appear to be used (yet) - is there a reason for that?

Comment (KPrasch) [reply tests/utils/blockchain.py:2]: if it's unused feel free to remove it. I think I was using it as a type hint in an earlier iteration.

**Review by vzotova — APPROVED** [review]

**Review by theref — APPROVED** [review]

**Pull request merged**